[quant][refactor] Add `get_qparams` and revert the changes to `calculate_qparams`

jerryzh168 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#30262 [quant][refactor] Add `get_qparams` and revert the changes to `calculate_qparams`**

Summary:
`get_qparams` returns all parameters that's needed to call quantize function

Test Plan:
python test/test_jit.py

Reviewers:
pt1quant

Subscribers:

Tasks:

Tags:

Differential Revision: [D18645047](https://our.internmc.facebook.com/intern/diff/D18645047)

jerryzh168 committed:
[quant][refactor] Add `get_qparams` and revert the changes to `calculate_qparams`

Summary:
`get_qparams` returns all parameters that's needed to call quantize function

Test Plan:
python test/test_jit.py

Reviewers:
pt1quant

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][refactor] Add `get_qparams` and revert the changes to `calculate_qparams`"


Summary:
`get_qparams` returns all parameters that's needed to call quantize function

Test Plan:
python test/test_jit.py

Reviewers:
pt1quant

Subscribers:

Tasks:

Tags:

Differential Revision: [D18645047](https://our.internmc.facebook.com/intern/diff/D18645047)

[ghstack-poisoned]

raghuramank100 commented on torch/csrc/jit/passes/quantization.cpp:
```diff
@@ -544,9 +544,9 @@ std::tuple<IValue, IValue> QuantizeHelper::getQParams(Value* v) {
-  auto calculate_qparams = om.get_method("calculate_qparams");
-  IValue qparams = calculate_qparams(std::vector<IValue>());
-  checkCalculateQParamsResult(qparams);
+  auto get_qparams = om.get_method("get_qparams");
```
NIT: Might be better to use a more descriptive name than 'om', perhaps: 'observer_module'

raghuramank100 commented on torch/csrc/jit/passes/quantization.cpp:
```diff
@@ -503,16 +503,16 @@ Value* QuantizeHelper::removeObserver(
   TORCH_CHECK(
       tp->elements().size() == 2 || tp->elements().size() == 3,
-      "`calculate_qparams` function is expected to reutrn a "
+      "`get_qparams` function is expected to reutrn a "
```
NIT: typo 'return'

raghuramank100 commented on torch/nn/quantized/modules/utils.py:
```diff
@@ -1,13 +1,13 @@
 import torch
 
 def _quantize_weight(float_wt, observer):
+    wt_scale, wt_zp = observer.calculate_qparams()
```
Would it be cleaner to do:
```python
q_params = observer.get_qparams()
if ....
    qweight = torch.quantize_per_tensor(float_wt, float(q_params[0]), int(q_params[1]), torch.qint8)
elif ...
    qweight = torch.quantize_per_tensor(float_wt, q_params[0].to(torch.double), q_params[1].to(torch.int64), q_params[2], torch.qint8)
```

jerryzh168 replied:
probably better to keep this way... I think maybe we don't need get_qparams, we can get access to ch_axis through observer module

raghuramank100 reviewed (approved):
Looks good, have a few suggestions.

jerryzh168 committed:
Update on "[quant][refactor] Add `get_qparams` and revert the changes to `calculate_qparams`"


Summary:
`get_qparams` returns all parameters that's needed to call quantize function

Test Plan:
python test/test_jit.py

Reviewers:
pt1quant

Subscribers:

Tasks:

Tags:

Differential Revision: [D18645047](https://our.internmc.facebook.com/intern/diff/D18645047)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][refactor] Add `get_qparams` and revert the changes to `calculate_qparams`"


Summary:
`get_qparams` returns all parameters that's needed to call quantize function

Test Plan:
python test/test_jit.py

Reviewers:
pt1quant

Subscribers:

Tasks:

Tags:

Differential Revision: [D18645047](https://our.internmc.facebook.com/intern/diff/D18645047)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][refactor] Add `get_qparams` and revert the changes to `calculate_qparams`"


Summary:
`get_qparams` returns all parameters that's needed to call quantize function

Test Plan:
python test/test_jit.py

Reviewers:
pt1quant

Subscribers:

Tasks:

Tags:

Differential Revision: [D18645047](https://our.internmc.facebook.com/intern/diff/D18645047)

[ghstack-poisoned]

jerryzh168 commented:
I'll remove `get_qparams` in another PR, we can refactor the code of getting qparams from observer modules later

jerryzh168 committed:
Update on "[quant][refactor] Add `get_qparams` and revert the changes to `calculate_qparams`"


Summary:
`get_qparams` returns all parameters that's needed to call quantize function

Test Plan:
python test/test_jit.py

Reviewers:
pt1quant

Subscribers:

Tasks:

Tags:

Differential Revision: [D18645047](https://our.internmc.facebook.com/intern/diff/D18645047)

[ghstack-poisoned]
